src/core/Segments.tsx

```diff
@@ -11,24 +11,21 @@ type SegmentsProps = {
 }
 
 type Api = {
-  subscribe: (ref) => void
+  subscribe: (ref: React.RefObject<SegmentObject>) => void
 }
 
-type Segment = {
-  start: THREE.Vector3
-  end: THREE.Vector3
-  color?: THREE.Color
+type SegmentRef = React.RefObject<SegmentObject>
+type SegmentProps = Omit<JSX.IntrinsicElements['segmentObject'], 'start' | 'end' | 'color'> & {
+  start: ReactThreeFiber.Vector3
+  end: ReactThreeFiber.Vector3
+  color?: ReactThreeFiber.Color
 }
 
-type SegmentRef = React.RefObject<Segment>
-type SegmentProps = JSX.IntrinsicElements['segmentObject'] & Segment
-
 const context = React.createContext<Api>(null!)
 
-const arrColor = (color) => (color instanceof THREE.Color ? color.toArray() : color)
-const arrPos = (pos) => (pos instanceof THREE.Vector3 ? pos.toArray() : pos)
-
 const Segments = React.forwardRef<Line2, SegmentsProps>((props, forwardedRef) => {
+  React.useMemo(() => extend({ SegmentObject }), [])
```

Comment:
this doesn't feel right, why wouldn't we extend in the global space? cc @drcmda @CodyJasonBennett

Reply:
We do that also in the Instancing component

Reply:
https://github.com/pmndrs/drei/blob/bc6e3b0ddc5078cdd77e6385ffad0efa45dbf4d1/src/core/Instances.tsx#L39

Reply:
Furthermore, we do that for each instance